tools/toolchains/grpc_ratls/ra_tls/grpc/v1.38.1/examples/cpp/ratls/grpc_ratls_client.cc

```diff
@@ -128,21 +136,75 @@ int grpc_ratls_get_secret(
     GrSecretClient gr_secret(channel);
 
     std::string secret = gr_secret.GetSecret(name);
-    //std::cout << "secret received: " << secret << std::endl;
+    // std::cout << "secret received: " << secret << "len: " << secret.length() << std::endl;
+
+    if (secret.empty()) {
+        return GRPC_RATLS_NO_SECRET;
+    } else {
+        //Decode From Base64
+        size_t len = base64_decode_len(secret.c_str());
+        if (len) {
+            char *secret_orig = (char *)malloc(len);
+            if (!secret_orig) {
+                return GRPC_RATLS_BUF_ERR;
+            }
+            base64_decode(secret.c_str(), (unsigned char *)secret_orig, len);
+            secret_string->assign(secret_orig, secret_orig + len - 1);
+            free(secret_orig);
 
-    //Decode From Base64
-    size_t len = base64_decode_len(secret.c_str());
-    if (len) {
-        char *secret_orig = (char *)malloc(len);
-        base64_decode(secret.c_str(), (unsigned char *)secret_orig, len);
-        std::string secret_string(secret_orig, secret_orig + len - 1);
+            return GRPC_RATLS_SUCCESS;
+        }
+
+        return GRPC_RATLS_ERR;
+    }
+}
+
+// Get secret to file
+int grpc_ratls_get_secret(
+    const char *server_addr,
+    const char *config_json,
+    const char *name,
+    const char *secret_file
+)
+{
+    std::string secret_string;
+    grpc_ratls_result_t ret = grpc_ratls_get_secret_string(
+        server_addr, config_json, name, &secret_string
+    );
 
+    if (ret == GRPC_RATLS_SUCCESS) {
         //write to file
         std::ofstream myfile;
         myfile.open(secret_file);
         myfile << secret_string;
         myfile.close();
     }
 
-    return 0;
+    return ret;
+}
+
+// Get secret to buffer
+int grpc_ratls_get_secret_to_buf(
+    const char *server_addr,
+    const char *config_json,
+    const char *name,
+    char *secret_buf,
+    unsigned int *buf_len
+)
+{
+    std::string secret_string;
+    grpc_ratls_result_t ret = grpc_ratls_get_secret_string(
+        server_addr, config_json, name, &secret_string
+    );
+
+    if (ret == GRPC_RATLS_SUCCESS) {
+        if (*buf_len < secret_string.size()) {
+            return GRPC_RATLS_BUF_ERR;
```

Comment:
If the input buffer is too small, it should output the actual required size from `buf_len` and return `GRPC_RATLS_BUF_TOO_SMALL`.